esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,181 @@
+    pt_lat1 = []
+    pt_lon1 = []
+
+    if not isinstance(pt_lat, list):
+        pt_lat1.append(pt_lat)
+        pt_lon1.append(pt_lon)
+
+    else:
+        for n_lat in np.arange(len(pt_lat)):
+            pt_lat1.append(pt_lat[n_lat])
+            pt_lon1.append(pt_lon[n_lat])
+
+    if len(pt_lat1) != len(pt_lon1):
+        raise AeroAnsError('Extract_pt:Mismatch in number of lat/long values')
+
+    if 'level' in kwargs and 'height' in kwargs:
+        raise AeroAnsError('Extract_pt: Both Level and Height requested')
+
+    if icube.coords()[0].name() == 'time':  # only lat,lon,lev accepted
+        raise AeroAnsError(
+            'Extract_pt:Cannot handle time dimension at present')
+
+    if 'level' in kwargs and not icube.coord(
+            'model_level_number') and not icube.coord('pseudo_level'):
+        raise AeroAnsError('Extract_pt:Level requested, but not found in cube')
+
+    if 'height' in kwargs:
+        pt_hgt = []
+
+        if not isinstance(kwargs['height'], list):
+            pt_hgt.append(kwargs['height'])
+
+        else:
+            pt_hgt.extend(kwargs['height'])
+
+        if len(pt_lat1) != len(pt_hgt):
+            raise AeroAnsError(
+                'Extract_pt:Mismatch in number of points for lat/long/height')
+
+        # Check that heights have been merged with orography.
+        if not icube.coords('altitude'):
+            raise AeroAnsError(
+                'Extract_pt:Height requested but input data does not contain \
+                "Altitude" coordinate')
+
+        # Store the min and max altitudes from cube data so that user
+        # cannot request points located below/ above that.
+        # Will extract =min/max if beyond limits.
+        hgt_min = icube.coord('altitude').points.min()
+        hgt_max = icube.coord('altitude').points.max()
+
+    # ---------- Finished checks -- begin processing -------------------------
+
+    # If level specified, extract slice first
+    if 'level' in kwargs:
+
+        if icube.coord('model_level_number'):
+            icube = icube.extract(
+                iris.Constraint(model_level_number=kwargs['level']))
+
+        else:
+
+            if icube.coord('pseudo_level'):
+
+                icube = icube.extract(
+                    iris.Constraint(pseudo_level=kwargs['level']))
+
+    # Extract values for specified points lat/lon
+
+    # ******** Does not seem to handle multiple points if 3-D *************
+
+    data_out = []
+
+    for n_lat1 in np.arange(len(pt_lat1)):
+        pt_coords = [('latitude', pt_lat1[n_lat1]),
+                     ('longitude', pt_lon1[n_lat1])]
+
+        if 'nearest' in kwargs and kwargs['nearest']:
+            tcube = icube.interpolate(pt_coords, iris.analysis.Nearest())
+
+        else:
+            tcube = icube.interpolate(pt_coords, iris.analysis.Linear())
+
+        # Extract at requested height
+
+        if 'height' in kwargs:
+            point = max(pt_hgt[n_lat1], hgt_min)
```

Comment:
this is the only point raised by Codacy that I actually care for (the rest are all blank lines needed or not, and too many variables or statements in funcs - they are not mandatory) - this would be nice to be compacted with the block up a few lines where another check on `height` in kwargs is performed, for better readibility of the code - not a bug, it's just nicer for someone else reading the code

Reply:
Done. (I think). I shifted things around a bit so the height coord is now set when the checks are being done. Similarly I have now set the lat/lon coord with the lat/lon checks.